src/libcore/sync/atomic.rs

```diff
@@ -405,6 +404,7 @@ impl AtomicBool {
     #[inline]
     #[stable(feature = "rust1", since = "1.0.0")]
     pub fn load(&self, order: Ordering) -> bool {
+        // SAFETY: data races are prevented by atomic intrinsics
```

Comment:
I believe the reason this needs an unsafe block is the passing of the raw pointer, and we know that raw pointer is valid because we got it from a reference

Reply:
Just noticed this. Would saying something like

> SAFETY: any data races are prevented by atomic intrinsics, and the raw pointer passed in is valid because we got it from a reference

be better? Or should we just leave out the first bit about atomics preventing data races?

Reply:
Atomic and non-atomic accesses still race with each other, right?

Reply:
That's true, but I think in the context of AtomicBool, any accesses are going to be atomic or unique because of `mut` access, which is why it's safe.